core/src/main/scala/org/bitcoins/core/currency/CurrencyUnits.scala

```diff
@@ -6,15 +6,52 @@ import org.bitcoins.core.protocol.NetworkElement
 import org.bitcoins.core.serializers.RawSatoshisSerializer
 import org.bitcoins.core.util.Factory
 import scodec.bits.ByteVector
+import scala.math.Numeric
 
 sealed abstract class CurrencyUnit
     extends NetworkElement
     with Ordered[CurrencyUnit]
-    with BasicArithmetic[CurrencyUnit] {
+    with BasicArithmetic[CurrencyUnit]
+    with Numeric[CurrencyUnit] {
   type A
 
   def satoshis: Satoshis
 
+  override def plus(x: CurrencyUnit, y: CurrencyUnit): CurrencyUnit =
```

Comment:
Your effictively implementing all of these twice. Can you delegate to the other ones? i.e. 

```scala
  override def +(c: CurrencyUnit): CurrencyUnit = {
    plus(this,c)
  }
```